src/cargo/util/config.rs

```diff
@@ -21,6 +22,30 @@ use util::toml as cargo_toml;
 
 use self::ConfigValue as CV;
 
+#[derive(PartialEq, Eq, Hash)]
+struct ConfigFile {
+    path: Option<PathBuf>,
+}
+
+impl ConfigFile {
+    pub fn new(path: PathBuf) -> ConfigFile {
+        let canonical = match fs::canonicalize(path) {
```

Comment:
No, it's new, but It was suggested in the original issue (here ) and I thought it was a good idea. I can remove it of course.